server/delay_server/src/irodsDelayServer.cpp

```diff
@@ -676,6 +677,51 @@ namespace
     } // make_delay_queue_query_processor
 } // anonymous namespace
 
+bool is_current_server_leader(void) {
+    std::string hostname;
+    char hn[HOST_NAME_MAX + 1]{};
+
+    if (gethostname(hn, sizeof(hn)) != 0) {
+        logger::delay_server::warn("is_current_server_leader: Failed to retrieve local server's hostname.");
+        return false;
+    }
+
+    if (const auto hn_resolved = resolve_hostname(hn, hostname_resolution_scheme::match_preferred); hn_resolved) {
+        hostname = *hn_resolved;
+    }
+    else {
+        hostname = std::string{hn};
+    }
+    auto conn = get_new_connection(std::nullopt);
```

Comment:
Consider constructing the client connection directly.

That makes the code clearer and avoids the unnecessary checks in `get_new_connection`. It also guards against `get_new_connection` changing in behavior and possibly breaking your code.
```cpp
irods::experimental::client_connection conn;
```